pmlc/conversion/SCFToGPU/SCFToGPUPass.cpp

```diff
@@ -29,6 +29,88 @@ using namespace mlir::scf; // NOLINT
 namespace pmlc::conversion::scf_to_gpu {
 
 namespace {
+// Change new added SCF to standards and avoid influence on gpu::LaunchOp
+// content
+struct ForLowering : public OpRewritePattern<mlir::scf::ForOp> {
+  using OpRewritePattern<mlir::scf::ForOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mlir::scf::ForOp forOp,
+                                PatternRewriter &rewriter) const override;
+};
+
+// From LowerToCFGPass, avoid conversion in gpu::LaunchOp contents.
+LogicalResult ForLowering::matchAndRewrite(scf::ForOp forOp,
+                                           PatternRewriter &rewriter) const {
+  Operation *nested = &forOp.getBody()->front();
+  auto parentOp = forOp.getParentOp();
+  if (dyn_cast<scf::ForOp>(nested) || isa<scf::ForOp>(*parentOp)) {
```

Comment:
This whole thing is added to turn single/unnested for loops to standard and then run on host side. I think maybe a new pass need to be created to turn single for loops and ops outside of any for loops to 2 nested for loops , so they can be turned into a gpu module and run on GPU instead of running on host side.